WordPress/Sniffs/PHP/IniSetSniff.php

```diff
@@ -0,0 +1,173 @@
+	 *     $target_functions = array(
+	 *         (string) Function name.
+	 *     );
+	 */
+	protected $target_functions = array(
+		'ini_set' => array(),
+	);
+
+	/**
+	 * Array of options that are allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $whitelisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'valid_values' = array()
+	 *         )
+	 *     );
+	 */
+	protected $whitelisted_options = array(
+		'auto_detect_line_endings' => array(),
+		'highlight.bg'             => array(),
+		'highlight.comment'        => array(),
+		'highlight.default'        => array(),
+		'highlight.html'           => array(),
+		'highlight.keyword'        => array(),
+		'highlight.string'         => array(),
+		'short_open_tag'           => array(
+			'valid_values' => array( 'true', '1', 'On' ),
+		),
+	);
+
+	/**
+	 * Array of options that are not allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $blacklisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'invalid_values' = array()
+	 *             (string) 'message'
+	 *         )
+	 *     );
+	 */
+	protected $blacklisted_options = array(
+		'max_execution_time' => array(
+			'message' => 'Use `set_time_limit()` instead.',
+		),
+		'short_open_tag' => array(
+			'invalid_values' => array( 'false', '0', 'Off' ),
+			'message'        => 'Turning off short_open_tag is prohibited as it might possibily break other plugins.',
+		),
+		'bcmath.scale' => array(
+			'message' => 'Use `bcscale()` instead.',
+		),
+		'display_errors' => array(
+			'message' => 'Use `WP_DEBUG_DISPLAY` instead.',
+		),
+		'error_reporting' => array(
+			'message' => 'Use `WP_DEBUG` instead.',
+		),
+		'filter.default' => array(
+			'message' => 'Use the filter flag constants when calling the functions instead (as you will possibly break other plugins if you change this).',
+		),
+		'filter.default_flags' => array(
+			'message' => 'Use the filter flag constants when calling the functions instead (as you will possibly break other plugins if you change this).',
+		),
+		'iconv.input_encoding' => array(
+			'message' => 'PHP < 5.6 only - use `iconv_set_encoding()` instead.',
+		),
+		'iconv.internal_encoding' => array(
+			'message' => 'PHP < 5.6 only - use `iconv_set_encoding()` instead.',
+		),
+		'iconv.output_encoding' => array(
+			'message' => 'PHP < 5.6 only - use `iconv_set_encoding()` instead.',
+		),
+		'ignore_user_abort' => array(
+			'message' => 'Use `ignore_user_abort()` instead.',
+		),
+		'log_errors' => array(
+			'message' => 'Use `WP_DEBUG_LOG` instead.',
+		),
+		'memory_limit' => array(
+			'message' => 'Use `wp_raise_memory_limit()` or hook into the filters in that function.',
+		),
+	);
+
+	/**
+	 * Process the parameter of a matched function. Errors if an option
```

Comment:
Suggestion: Start the phrase from `Errors if ..` on a new line as the long description (with a blank line between the short and the long description).